src/integrationTest/resources/TestProject/build.gradle

```diff
@@ -21,18 +21,23 @@ apply plugin: 'java'
 subprojects {
   apply plugin: "umple.gradle.plugin"
   apply plugin: "java"
-  
+
+// TODO this should not be neccesary. Find out why compileUmple is returning "UP-TO-DATE" when the output files have been deleted. 
+compileUmple {
```

Comment:
We should get rid of this before merging this PR.

Reply:
This TODO should be fixed by using [IncrementalTaskInputs](https://docs.gradle.org/3.1/javadoc/org/gradle/api/tasks/incremental/IncrementalTaskInputs.html). You can see how I used them here: https://github.com/assertj/assertj-generator-gradle-plugin/blob/9f0ec19e910a3d7a4add5de461481454688063d8/src/main/groovy/org/assertj/generator/gradle/tasks/AssertJGenerationTask.groovy#L67

Reply:
I'm not sure if IncrementalTaskInputs are the answer. The problem isn't that gradle is mistaking a modified/removed input file for an unmodified input file, the problem is that gradle isn't checking whether generated/output files (i.e. java files) exist when it does it's "up to date" check. We really just need gradle to check to see if the output files have been removed, and if they have, to rerun the generation task.

Reply:
That's what IncrementalTaskInputs does. If you register all of the produced files with ITI, it will remove/update them when required.